### ISSUE TITLE
venv-update is uninstalling pkg_resources inside tox

### ISSUE THREAD

**Issue by chriskuehl** [Yelp/venv-update#186]
This is really weird. On Debian stretch, I'm seeing venv-update uninstall pkg_resources, but only in tox.

Here's a reproduction done in the `debian:stretch` docker image:

1. `apt-get update && apt-get -y install ca-certificates python python3.5 python-tox curl nano`
2. `curl -o venv-update https://raw.githubusercontent.com/Yelp/venv-update/master/venv_update.py && chmod +x venv-update`
3. make a `tox.ini` with these contents:

```ini
[tox]
envlist = py35
skipsdist = true

[testenv]
venv_update =
    {toxinidir}/venv-update \
        venv= {envdir} \
        install= pytest
commands =
    {[testenv]venv_update}
    python -c 'import pkg_resources'
```

4. run `tox -e py35`

This produces the output:

```
root@10cff4432457:/# tox -e py35
py35 create: /.tox/py35
py35 installed: appdirs==1.4.0,packaging==16.8,pkg-resources==0.0.0,six==1.10.0
py35 runtests: PYTHONHASHSEED='3310243166'
py35 runtests: commands[0] | ///venv-update venv= /.tox/py35 install= pytest
> virtualenv /.tox/py35
Keeping valid virtualenv from previous run.
> pip install venv-update==2.1.0
Collecting venv-update==2.1.0
  Using cached venv_update-2.1.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=0.8.0 in /.tox/py35/lib/python3.5/site-packages (from venv-update==2.1.0)
Requirement already satisfied: wheel>0.25.0 in /.tox/py35/lib/python3.5/site-packages (from venv-update==2.1.0)
Requirement already satisfied: pip>=8.1.0 in /.tox/py35/lib/python3.5/site-packages (from venv-update==2.1.0)
Requirement already satisfied: packaging>=16.8 in /.tox/py35/lib/python3.5/site-packages (from setuptools>=0.8.0->venv-update==2.1.0)
Requirement already satisfied: six>=1.6.0 in /.tox/py35/lib/python3.5/site-packages (from setuptools>=0.8.0->venv-update==2.1.0)
Requirement already satisfied: appdirs>=1.4.0 in /.tox/py35/lib/python3.5/site-packages (from setuptools>=0.8.0->venv-update==2.1.0)
Installing collected packages: venv-update
Successfully installed venv-update-2.1.0
> pip-faster install --upgrade --prune pytest
Collecting pytest
  slow: full search for unpinned requirement pytest
  Using cached pytest-3.0.6-py2.py3-none-any.whl
slow: full search for unpinned requirement setuptools in /.tox/py35/lib/python3.5/site-packages (from pytest)
Requirement already up-to-date: setuptools in /.tox/py35/lib/python3.5/site-packages (from pytest)
Collecting py>=1.4.29 (from pytest)
  slow: full search for unpinned requirement py>=1.4.29 (from pytest)
  Using cached py-1.4.32-py2.py3-none-any.whl
slow: full search for unpinned requirement packaging>=16.8 in /.tox/py35/lib/python3.5/site-packages (from setuptools->pytest)
Requirement already up-to-date: packaging>=16.8 in /.tox/py35/lib/python3.5/site-packages (from setuptools->pytest)
slow: full search for unpinned requirement six>=1.6.0 in /.tox/py35/lib/python3.5/site-packages (from setuptools->pytest)
Requirement already up-to-date: six>=1.6.0 in /.tox/py35/lib/python3.5/site-packages (from setuptools->pytest)
slow: full search for unpinned requirement appdirs>=1.4.0 in /.tox/py35/lib/python3.5/site-packages (from setuptools->pytest)
Requirement already up-to-date: appdirs>=1.4.0 in /.tox/py35/lib/python3.5/site-packages (from setuptools->pytest)
Installing collected packages: py, pytest
Successfully installed py-1.4.32 pytest-3.0.6
> pip uninstall --yes pkg-resources
Uninstalling pkg-resources-0.0.0:
  Successfully uninstalled pkg-resources-0.0.0
py35 runtests: commands[1] | python -c import pkg_resources
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'pkg_resources'
ERROR: InvocationError: '/.tox/py35/bin/python -c import pkg_resources'
____________ summary ____________
ERROR:   py35: commands failed
```

but... why is it uninstalling `pkg-resources`?

This only appears to happen when using tox (calling the same command with `venv-update` directly doesn't uninstall it). This also doesn't happen on Debian jessie (with a deadsnakes backport of python3.5).

**Comment by asottile** [comment]
Ah this is due to debian's weird packaging of pkg-resources.  Normally the `pkg_resources` module is provided by setuptools but debian has decided to split them apart.

I guess we should just add pkg-resources to this list? https://github.com/Yelp/venv-update/blob/4244b3f32fa7621b0bafeda81989f082545f2182/pip_faster.py#L387-L392

**Comment by asottile** [comment]
I can also reproduce with:

```bash
docker run -ti -v $PWD/setup.py:/setup.py:ro -v $PWD/pip_faster.py:/pip_faster.py:ro -v $PWD/venv_update.py:/venv_update.py:ro ubuntu:xenial bash -xc '
    apt update &&
    apt install -y virtualenv &&
    virtualenv venv -ppython3 &&
    venv/bin/pip install venv-update &&
    venv/bin/pip-faster install six --prune &&
    venv/bin/python -c "import pkg_resources"
'
```

```
...
+ venv/bin/pip-faster install six --prune
Requirement already satisfied: six in /venv/lib/python3.5/site-packages
> pip uninstall --yes pkg-resources
Uninstalling pkg-resources-0.0.0:
  Successfully uninstalled pkg-resources-0.0.0
+ venv/bin/python -c 'import pkg_resources'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named 'pkg_resources'
```

With my patch (that I'll PR in a sec):

```
...
+ venv/bin/pip-faster install six --prune
Error [Errno 2] No such file or directory: 'git' while executing command git rev-parse
Requirement already satisfied: six in /venv/lib/python3.5/site-packages
Error [Errno 2] No such file or directory: 'git' while executing command git rev-parse
+ venv/bin/python -c 'import pkg_resources'
```

**Comment by chriskuehl** [comment]
Fixed in 2.1.1
